### PR TITLE
WIP feat(datasets): added unlink command

### LINKED ISSUE #447: cli: add unlink command
Add unlink command to unlink single or multiple files by pattern from dataset.

Example:
```console
$ renku dataset unlink mydataset --include=file* --exclude=file3
Warning: You are about to delete following files from "my-dataset" dataset:

  [red]
  /home/sam/Work/old-repo-lfs/data/my-dataset/test-data/file2
  /home/sam/Work/old-repo-lfs/data/my-dataset/test-data/file1
  [/red]

Do you wish to proceed? [y/N]: 
```

Optional flags:
--yes (automatically verifies deletion)
--verbose (shows successfully deleted files, after deletion)

Default include parameter is '*' (all files), and exclude None.

related #418

### PR DESCRIPTION
**Pull request by jsam**
**IMPORTANT: DO NOT MERGE**

Adds unlink command.

Fixes #447

### CODE REVIEW

**Comment by rokroskar** [comment]
Adding the `wip` to the title is enough to prevent merging. Or you can also open a draft pr. 😉

File: renku/api/datasets.py [hunk 1]
```diff
@@ -113,6 +113,27 @@ def with_dataset(self, name=None):
             with path.open('w') as f:
                 yaml.dump(source, f, default_flow_style=False)
 
+    def data_to_unlink(self, dataset, include, exclude=None):
```
Comment (jirikuncar) [inline renku/api/datasets.py]:
```suggestion
    def data_to_unlink(self, dataset, include=None, exclude=None):
```

File: renku/api/datasets.py [hunk 2]
```diff
@@ -436,3 +457,50 @@ def check_for_git_repo(url):
...
+    :param collection: Arbitary collection, same in format to dataset.files.
+    """
+    collection = collection or dataset.files
+
+    result = {}
+    for path_, file_ in collection.items():
+        key = os.path.realpath(
+            str(client.renku_datasets_path / dataset.identifier.hex / path_)
+        )
+        file_.dataset = dataset.name
```
Comment (jirikuncar) [inline renku/api/datasets.py]: We can remove the `.dataset` property and use `.__reference__` instead.

File: renku/cli/dataset.py [hunk 3]
```diff
@@ -157,6 +163,244 @@ def add(client, name, urls, nocopy, relative_to, target, force):
         raise BadParameter('Could not process {0}'.format(url))
 
 
+@dataset.command('ls-files')
+@click.option(
+    '--dataset', multiple=True, help='Filter files in specific dataset.'
```
Comment (jirikuncar) [inline renku/cli/dataset.py]: aren't we using argument for `dataset` name?

File: renku/cli/dataset.py [hunk 4]
```diff
@@ -157,6 +163,244 @@ def add(client, name, urls, nocopy, relative_to, target, force):
...
+)
+@click.option(
+    '--authors',
+    help='Filter files which where authored by specific authors. '
+    'Multiple authors are specified by comma.'
+)
+@click.option(
+    '--include',
```
Comment (jirikuncar) [inline renku/cli/dataset.py]: please add `-I`

File: renku/cli/dataset.py [hunk 5]
```diff
@@ -157,6 +163,244 @@ def add(client, name, urls, nocopy, relative_to, target, force):
...
+    default='*',
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
    default=None,
```

File: renku/cli/dataset.py [hunk 6]
```diff
@@ -157,6 +163,244 @@ def add(client, name, urls, nocopy, relative_to, target, force):
...
+    multiple=True,
+    help='Include files matching given pattern.'
+)
+@click.option(
+    '--exclude', multiple=True, help='Exclude files matching given pattern.'
```
Comment (jirikuncar) [inline renku/cli/dataset.py]: please add `-X`

File: renku/errors.py [hunk 7]
```diff
@@ -258,3 +258,15 @@ def __init__(self, repo_path):
             'Initialize it with "renku init"'.format(repo_path=repo_path)
         )
         super(UninitializedProject, self).__init__(msg)
+
+
+class ResourceNotFound(RenkuException, click.ClickException):
```
Comment (jirikuncar) [inline renku/errors.py]: can we reuse `NotFound` error? it is probably not used anymore

**Pull request closed without merging**